src/components/Bookmark.js

```diff
@@ -5,29 +5,18 @@ import { FaRegBookmark as InactiveBookmark } from 'react-icons/fa';
 import { UserSpellbookContext } from '../contexts/UserSpellbookContext';
 
 const Bookmark = ({ spell }) => {
-  const [ isBookmarked, setIsBookmarked ] = useState(false);
-  const { bookmarks, setBookmarks } = useContext(UserSpellbookContext);
+  const { bookmarks, dispatch } = useContext(UserSpellbookContext);
```

Comment:
I was scared of reducers, but I'm glad you went for it! The result is much cleaner (and actually works lol).

Reply:
They are confusing, but I think I'm starting to get the hang of them!